File: snippets/screens/compose/welcome_screen/1/welcome_screen.kt
package com.aiq.screens

import androidx.compose.animation.core.*
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.scale
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.delay

@Composable
fun WelcomeScreen(
    onStartTutorial: () -> Unit = {},
    onSkipTutorial: () -> Unit = {}
) {
    var titleOpacity by remember { mutableStateOf(0f) }
    var contentOffset by remember { mutableStateOf(30.dp) }
    var illustrationScale by remember { mutableStateOf(0.8f) }
    var buttonsOpacity by remember { mutableStateOf(0f) }

    // Animate values
    val titleOpacityAnim by animateFloatAsState(
        targetValue = titleOpacity,
        animationSpec = tween(600)
    )

    val contentOffsetAnim by animateDpAsState(
        targetValue = contentOffset,
        animationSpec = tween(800, easing = FastOutSlowInEasing)
    )

    val illustrationScaleAnim by animateFloatAsState(
        targetValue = illustrationScale,
        animationSpec = spring(
            dampingRatio = Spring.DampingRatioMediumBouncy,
            stiffness = Spring.StiffnessLow
        )
    )

    val buttonsOpacityAnim by animateFloatAsState(
        targetValue = buttonsOpacity,
        animationSpec = tween(600)
    )

    LaunchedEffect(Unit) {
        titleOpacity = 1f
        contentOffset = 0.dp
        delay(200)
        illustrationScale = 1f
        delay(400)
        buttonsOpacity = 1f
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color.White)
            .padding(24.dp)
    ) {
        Spacer(modifier = Modifier.height(40.dp))

        // Welcome header
        Column(
            modifier = Modifier
                .graphicsLayer {
                    alpha = titleOpacityAnim
                    translationY = contentOffsetAnim.toPx()
                }
        ) {
            // Welcome badge
            Surface(
                shape = RoundedCornerShape(20.dp),
                color = Color(0xFF6366F1).copy(alpha = 0.1f)
            ) {
                Text(
                    text = "Welcome!",
                    color = Color(0xFF6366F1),
                    fontSize = 14.sp,
                    fontWeight = FontWeight.SemiBold,
                    modifier = Modifier.padding(horizontal = 12.dp, vertical = 6.dp)
                )
            }

            Spacer(modifier = Modifier.height(24.dp))

            Text(
                text = "Ready to test your\nAI intelligence?",
                fontSize = 32.sp,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF1F2937),
                lineHeight = 38.sp
            )

            Spacer(modifier = Modifier.height(16.dp))

            Text(
                text = "Join thousands of users discovering their AI potential. Let's get you started on your journey to understanding artificial intelligence.",
                fontSize = 16.sp,
                color = Color(0xFF6B7280),
                lineHeight = 24.sp
            )
        }

        Spacer(modifier = Modifier.weight(1f))

        // Illustration
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .scale(illustrationScaleAnim),
            contentAlignment = Alignment.Center
        ) {
            Card(
                modifier = Modifier
                    .size(200.dp)
                    .shadow(
                        elevation = 40.dp,
                        shape = CircleShape,
                        ambientColor = Color(0xFF6366F1).copy(alpha = 0.3f)
                    ),
                shape = CircleShape,
                colors = CardDefaults.cardColors(
                    containerColor = Color.Transparent
                )
            ) {
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .background(
                            brush = Brush.radialGradient(
                                colors = listOf(
                                    Color(0xFF6366F1),
                                    Color(0xFF8B5CF6)
                                )
                            )
                        ),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Default.Psychology,
                        contentDescription = "AI Brain",
                        tint = Color.White,
                        modifier = Modifier.size(80.dp)
                    )
                }
            }
        }

        Spacer(modifier = Modifier.weight(1f))

        // Action buttons
        Column(
            modifier = Modifier.graphicsLayer { alpha = buttonsOpacityAnim },
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Button(
                onClick = onStartTutorial,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = Color(0xFF6366F1)
                ),
                shape = RoundedCornerShape(16.dp)
            ) {
                Text(
                    text = "Start Tutorial",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.White
                )
                Spacer(modifier = Modifier.width(8.dp))
                Icon(
                    imageVector = Icons.Default.ArrowForward,
                    contentDescription = null,
                    modifier = Modifier.size(20.dp),
                    tint = Color.White
                )
            }

            OutlinedButton(
                onClick = onSkipTutorial,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp),
                border = BorderStroke(1.dp, Color(0xFFE5E7EB)),
                shape = RoundedCornerShape(16.dp)
            ) {
                Text(
                    text = "Skip Tutorial",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Medium,
                    color = Color(0xFF6B7280)
                )
            }
        }
    }
}

@Preview(showBackground = true)
@Composable
fun WelcomeScreenPreview() {
    WelcomeScreen()
}

File: snippets/screens/swiftui/welcome_screen/3/welcome_screen.swift
import SwiftUI

struct WelcomeScreen: View {
    @State private var titleOpacity: Double = 0
    @State private var contentOffset: CGFloat = 30
    @State private var illustrationScale: CGFloat = 0.8
    @State private var buttonsOpacity: Double = 0

    var body: some View {
        VStack {
            VStack(alignment: .leading, spacing: 24) {
                // Welcome badge
                HStack {
                    HStack {
                        Text("Welcome!")
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundColor(Color(red: 0.39, green: 0.40, blue: 0.95))
                    }
                    .padding(.horizontal, 12)
                    .padding(.vertical, 6)
                    .background(Color(red: 0.39, green: 0.40, blue: 0.95).opacity(0.1))
                    .cornerRadius(20)

                    Spacer()
                }

                // Main title
                VStack(alignment: .leading, spacing: 16) {
                    Text("Ready to test your\nAI intelligence?")
                        .font(.system(size: 32, weight: .bold))
                        .foregroundColor(Color(red: 0.12, green: 0.16, blue: 0.22))
                        .lineSpacing(4)

                    Text("Join thousands of users discovering their AI potential. Let's get you started on your journey to understanding artificial intelligence.")
                        .font(.system(size: 16))
                        .foregroundColor(Color(red: 0.42, green: 0.45, blue: 0.50))
                        .lineSpacing(4)
                }
            }
            .opacity(titleOpacity)
            .offset(y: contentOffset)
            .padding(.horizontal, 24)
            .padding(.top, 40)

            Spacer()

            // Illustration
            ZStack {
                Circle()
                    .fill(
                        LinearGradient(
                            gradient: Gradient(colors: [
                                Color(red: 0.39, green: 0.40, blue: 0.95),
                                Color(red: 0.55, green: 0.36, blue: 0.96)
                            ]),
                            startPoint: .topLeading,
                            endPoint: .bottomTrailing
                        )
                    )
                    .frame(width: 200, height: 200)
                    .shadow(color: Color(red: 0.39, green: 0.40, blue: 0.95).opacity(0.3), radius: 40, x: 0, y: 20)

                Image(systemName: "brain.head.profile")
                    .font(.system(size: 80))
                    .foregroundColor(.white)
            }
            .scaleEffect(illustrationScale)

            Spacer()

            // Action buttons
            VStack(spacing: 16) {
                Button(action: startTutorial) {
                    HStack {
                        Text("Start Tutorial")
                            .font(.system(size: 16, weight: .semibold))

                        Image(systemName: "arrow.right")
                            .font(.system(size: 16, weight: .semibold))
                    }
                    .foregroundColor(.white)
                    .frame(maxWidth: .infinity)
                    .frame(height: 56)
                    .background(Color(red: 0.39, green: 0.40, blue: 0.95))
                    .cornerRadius(16)
                }

                Button(action: skipTutorial) {
                    Text("Skip Tutorial")
                        .font(.system(size: 16, weight: .medium))
                        .foregroundColor(Color(red: 0.42, green: 0.45, blue: 0.50))
                        .frame(maxWidth: .infinity)
                        .frame(height: 56)
                        .overlay(
                            RoundedRectangle(cornerRadius: 16)
                                .stroke(Color(red: 0.90, green: 0.91, blue: 0.92), lineWidth: 1)
                        )
                }
            }
            .opacity(buttonsOpacity)
            .padding(.horizontal, 24)
            .padding(.bottom, 24)
        }
        .background(Color.white)
        .onAppear {
            startAnimations()
        }
    }

    private func startAnimations() {
        // Title animation
        withAnimation(.easeOut(duration: 0.6)) {
            titleOpacity = 1
            contentOffset = 0
        }

        // Illustration animation
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
            withAnimation(.spring(response: 0.8, dampingFraction: 0.6, blendDuration: 0)) {
                illustrationScale = 1.0
            }
        }

        // Buttons animation
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.4) {
            withAnimation(.easeOut(duration: 0.6)) {
                buttonsOpacity = 1
            }
        }
    }

    private func startTutorial() {
        // Navigate to tutorial
        // NavigationManager.shared.navigate(to: .tutorial)
    }

    private func skipTutorial() {
        // Navigate to main app
        // NavigationManager.shared.navigate(to: .main)
    }
}

#Preview {
    WelcomeScreen()
}